sdk/src/main/kotlin/io/rover/sdk/data/events/EventAnalyticsService.kt

```
@@ -0,0 +1,125 @@
+            put("timestamp", dateAsIso8601(Date()))
+            put("properties", eventInformation.attributes.encodeJson())
+        }.toString()
+    }
+
+    private fun sendRequest(eventInformation: EventEmitter.Event) {
+        val urlRequest = buildRequest(URL(ANALYTICS_ENDPOINT), accountToken)
+        val bodyData = encodeBody(eventInformation)
+
+        request(urlRequest, bodyData)
+    }
+
+    init {
+        eventEmitter.trackedEvents.subscribe { sendRequest(it) }
+    }
+
+    private fun dateAsIso8601(date: Date): String {
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSXXX", Locale.US).format(date)
+        } else {
+            // On legacy Android, we are using the RFC 822 (email) vs ISO 8601 date format, and
+            // we use the following regex to transform it to something 8601 compatible.
+            SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ", Locale.US).format(date)
+                .replace(Regex("(\\d\\d)(\\d\\d)$"), "$1:$2")
+        }
+    }
+
+    open fun request(
+        request: HttpRequest,
+        bodyData: String
+    ) {
+        AsyncTask.execute {
+            val connection = request.url.openConnection() as HttpURLConnection
+            val requestBody = bodyData.toByteArray(Charsets.UTF_8)
+
+            connection.apply {
+                    setFixedLengthStreamingMode(requestBody.size)
+                    request.headers.onEach { (field, value) -> setRequestProperty(field, value) }
+
+                    doOutput = true
+                    requestMethod = request.verb.wireFormat
+                }
+
+            try {
+                connection.outputStream.use { stream ->
+                    DataOutputStream(stream).use { dataOutputStream ->
+                        dataOutputStream.write(requestBody)
+                    }
+                }
+            } catch (e: IOException) {
+                return@execute
```

we should definitely print a log here using log.w.  I'm pretty sure silenced errors are prohibited by the Geneva convention.

---

are any other exceptions other than IOException possible here? We definitely do not want this class to ever crash customer apps.